developer/contributing.md

```diff
@@ -31,6 +31,84 @@ Start small and open up a dialogue with us. This will help to get your contribut
 -   Contributors should review the CLA.
 -   Code reviewed before merge acceptance.
 
+### Pull request template
+
+```
+Resolves #issueNumber
+Impact breaking|minor|bug|style|refactor
+
+## Issue
+
+Description of the issue this PR is solving, why it's happening and how it can be reproduced. This may differ from the original ticket as you now have more information at your disposal.
+
+## Solution
+
+Summarize the your solution to the problem, as well as any things you may have tried along the way that may not have worked.
+
+## Breaking changes
+
+- If you have a breaking changes, list them here, otherwise list none.
+
+## Testing
+
+1. Add steps to test this PR
+1. Be detailed enough that someone can work through it
+1. Avoid being too granular
+```
+
+#### Template sections
+
+##### Top section
+
+###### PR author
+
+The first line of every PR should have `Resolves #ticketNumber`.
+The second line of every PR should have `Impact breaking|minor|bug|style|refactor`.
+
+###### Reviewer
+
+Read over the original ticket, and understand the impact this PR will have on the codebase. If the PR doesn't follow the our template, reject and point the author of the PR to this doc.
```

Comment:
I'm confused by this section. Can we just have one PR Author and one PR Review section rather than these alternating ones? Am I the only one bothered by this? One is a list of drop-dead requirements and one is a list of "things to consider", I'm not sure if they are really equivalent?

Reply:
Somewhere (I can't tell where in this current format) we need to have a checklist of things code reviewers should consider.

1. Performance (does the code use best practices for minimizing the load on client and server hardware)
1. Concurrency (are data transactions multi-user safe? Am I prevented from accidentally overwriting someone else changes?)
1. Security (are permissions being checked everywhere appropriate? Are error messages being sanitized before they reach the client. Are we checking inputs properly wherever we accept client input?) 

These are not a comprehensive lists. But we should start compiling them.